Music.py

```diff
@@ -440,11 +545,30 @@ def rebuild_sequences(rom: Rom, sequences: list[Sequence], log: CosmeticsLog, sy
         rom.write_int32(0xB89AE0 + (i * 0x10), new_sequences[i].address)
         rom.write_int32(0xB89AE0 + (i * 0x10) + 0x04, new_sequences[i].size)
         seq = replacement_dict.get(i, None)
-        if seq:
-            rom.write_int16(0xB89AE0 + (i * 0x10) + 0x08, seq.type)
 
-    # Update instrument sets
-    for i in range(0x6E):
+    # Update instrument sets for bgm sequences
+    for i in bgmlist:
+        base = 0xB89911 + 0xDD + (i * 2)
+        j = replacement_dict.get(i if new_sequences[i].size else new_sequences[i].address, None)
+        if j:
+            rom.write_byte(base, j.instrument_set)
```

Comment:
And separately, what is this actually doing? The description of the previous PR only said that it removed some useless code changing the cache type, is this writing the correct cache type or something?

Reply:
no the code above is changing the sequence bank table which is what sets the instrument bank a sequence uses. the cachetype thing that i removed originally would make every sequences cache type 2 even though fanfares are supposed to be 1. the code was completely obsolete because the vanilla cache types are already set correctly and the original code would overwrite it with wrong data.